text/0000-stack-metadata.md

```diff
@@ -0,0 +1,95 @@
+# Meta
+[meta]: #meta
+- Name: Additional Stack Metadata
+- Start Date: 2020-05-12
+- Author(s): kvedurmu
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+We currently expose the Stack ID and Mixins via Labels on the build/run images. It would be great if we could leverage Labels to allow stack authors to specify additional image metadata (e.g OS Distro, Maintainer, Stack Version, etc.). The purpose of this RFC is to enumerate on additional metadata we want to expose in our stack images and standardize on these labels.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?  
+Exposing additional metadata in stack images will make it easier for end users to debug stack related issues. Also, by exposing this metadata, CNB platforms such as kpack can leverage this information to improve their UX. 
+- What is the expected outcome?  
+We converge on metadata we want to expose in stack images via labels, and all stack authors eventually support these.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This provides a high level overview of the feature.  
+
+- Define the target persona: stack author  
+- Explaining the feature largely in terms of examples.  
+The following labels will be added to stack images:  
+`io.buildpacks.stack.maintainer`: Name of Stack Maintainer  
+`io.buildpacks.stack.url`: URL for the stack (i.e repo link)  
+`io.buildpacks.stack.distro.name`: Name of OS Distribution  
+`io.buildpacks.stack.distro.version`: Version of OS Distribution  
+`io.buildpacks.stack.release`: Release Number
+`io.buildpacks.stack.release_date`: Release date of image
+`io.buildpacks.stack.description`: Description  
+
```

Comment:
Could we add a generic metadata field that is defined by the stack author?

E.g., `io.buildpacks.stack.metadata` could be defined by another out-of-spec RFC when the stack ID is io.buildpacks.stacks.bionic.

Reply:
Yeah that makes sense. I think that we should keep the other labels I proposed since we already use this pattern for the `stack.id` and `stack.mixins` and add this generic metadata label. How does that sound?

Reply:
Agreed on this.  I quite like the pattern of top-level keys being wholly reserved for the spec (i.e. `io.buildpacks.stack.*`) and having a key that is wholly reserved for users that the spec will never use (`io.buildpacks.stack.metadata`).  Top-level keys as they are described here look good.

Reply:
> E.g., `io.buildpacks.stack.metadata` could be defined by another out-of-spec RFC when the stack ID is io.buildpacks.stacks.bionic.

@sclevine can you elaborate on what you hope to have in there?

Reply:
> I quite like the pattern of top-level keys being wholly reserved for the spec (i.e. io.buildpacks.stack.*) and having a key that is wholly reserved for users that the spec will never use (io.buildpacks.stack.metadata).

While I agree in theory, we currently use a lot of `io.buildpacks.*.metadata` for our own purposes, rather than reserving them for users. (e.g `io.buildpacks.{lifecycle,build, builder}.metadata`)

Reply:
@ekcasey Well that's a kick in the shorts.  We can pick another key (`.user`) that represents the same thing, but alas is a naming inconsistency.  My priority stack is as follows:

1.  Designated area for all users data that we both agree to never spec inside of and that a user will never create outside of
2. Naming consistency

Even if that means we have some exceptions (e.g. `io.buildpacks{lifecycle,build,builder}.metadata`), I don't think we should introduce _new_ inconsistencies.

Reply:
@kvedurmu I don't think this one has been addressed yet.

Reply:
@nebhale oops just updated it.